docs/modules/architecture/pages/vector-search-overview.adoc

```diff
@@ -0,0 +1,122 @@
+
+Each collection is partitioned and replicated based on the system's general partitioning rules. Data partitioning is carried out using the collection key.
+
+NOTE: For more details about Hazelcast partitioning, please refer to the xref:data-partitioning.adoc[Data Partitioning and Replication].
+
+NOTE: Version 5.5/beta supports partitioning and migration but does not include support for the backup process.
+
+=== Data store
+Hazelcast stores data in-memory (RAM) for faster access. Presently, the only available data storage option is the JVM heap store.
+
+=== Fault Tolerance
+Hazelcast distributes storage data across all cluster members.
+In the event of a graceful shutdown, the data will be migrated to remaining active members.
+In version 5.5, there is no automatic data restoration in the event of an unexpected member loss.
+
+== Partitioned similarity search
+
+Vector indexes are partitioned, so when you execute similarity search all partitions need to be searched and partial results aggregated.
+It is important to understand the impact of that process on search performance and recall.
+
+=== Two-stage search
+
+The default search algorithm is two-stage search which works as follows:
+
+1. Member that received the query is a coordinator.
+2. Coordinator fans out search request to each member (including self). Each member is tasked with returning results from partitions it owns.
+3. Each member executes search on owned partitions in parallel and aggregates the partition results.
+4. Partially aggregated results are returned to coordinator.
+5. Coordinator aggregates partial results and generates final results.
+   If necessary, searches on some partitions can be retried individually, for example in case of migrations, members leaving the cluster or other errors.
+
+Aggregation at each stage is done based on score and only the best results are kept.
+
+There are 2 important parameters in this search algorithm that determine amount of data sent between the members and quality of the final result:
+
+- `partitionLimit` - number of search results obtained from each partition
+- `memberLimit` - number of search results returned from member to coordinator
+
+If `topK` denotes number of entries requested, then the following should hold in order for the system to be able to return enough results:
+
+- `partitionLimit * partitionCount >= topK`, `partitionLimit &lt;= topK`
+- `memberLimit * memberCount >= topK`, `memberLimit &lt;= topK`
+
+By default `partitionLimit` and `memberLimit` are equal to `topK` which satisfies the above inequalities but causes processing of much more results than the requested `topK`.
+This improves the overall quality of the results but can have a significant performance overhead because more entries are fetched from each partition of the index and sent between the members.
+
+NOTE: Consider tuning `partitionLimit` based on quality and latency requirements. It should be adjusted alongside number of partitions.
```

Comment:
```suggestion
NOTE: Consider tuning `partitionLimit` based on quality and latency requirements. Always consider the number of partitions when making adjustments.
```

Reply:
I meant than if `partitionLimit` and partition count should be tuned together, partition count not only should be considered but also changed (however there are more factors impacting partition count described in separate section)

Reply:
@k-jamroz amended and added reference to the partition count section. WDYT?